pkg/webhook/generic.go

```diff
@@ -0,0 +1,114 @@
+package webhook
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/Azure/brigade/pkg/brigade"
+	"github.com/Azure/brigade/pkg/storage"
+
+	restful "github.com/emicklei/go-restful"
+)
+
+type genericWebhook struct {
+	store storage.Store
+}
+
+type genericWebhookData struct {
+	Ref    string `json:"ref"`
+	Commit string `json:"commit"`
+}
+
+// NewGenericWebhook creates a go-restful handler for generic webhook.
+func NewGenericWebhook(s storage.Store) func(request *restful.Request, response *restful.Response) {
+	h := &genericWebhook{store: s}
+	return h.Handle
+}
+
+// Handle handles a generic webhook event.
+func (g *genericWebhook) Handle(request *restful.Request, response *restful.Response) {
+	log.Printf("Parameters: %s\n", request.PathParameters())
+
+	projectID := request.PathParameter("projectID")
+	secret := request.PathParameter("secret")
+
+	gwData := &genericWebhookData{}
+	err := request.ReadEntity(gwData)
+
+	if err != nil {
+		log.Printf("Failed to read GenericWebHookData: %s", err)
+		response.WriteErrorString(http.StatusBadRequest, "{\"status\": \"Malformed genericWebhookData\"}")
+		return
+	}
+
+	log.Printf("gwData: %#v\n", gwData)
+
+	body, err := ioutil.ReadAll(request.Request.Body)
```

Comment:
It might be good to move this part until after you load the project and compare the secret. Otherwise, you could end up with a denial of service-style attack where lots of anonymous clients send very large payloads to this endpoint.